bin/iop

```diff
@@ -55,6 +55,8 @@ function iop() {
         # Apply will make sure that any old config with the same tag that is no longer present will
         # be removed. The release MUST be unique across cluster.
         kubectl create  ns $ns > /dev/null  2>&1
+        #set label 'name' to namespace which is required if enableNamespacesByDefault is true. 
+        kubectl label namespace $ns name=$ns --overwrite
```

Comment:
From helm 2.10, helm will add new label `name={{ .Release.Namespace }}` to its created namespaces, see: https://github.com/helm/helm/pull/4231

And our current helm version `2.13.1`, is it an helm issue?

Reply:
that is only with tiller. we should not rely on tiller

and strongly prefer we don't add a label to the namespace and just fix the webhook config instead -- it's broken in istio/istio for the same reason, there is an open issue

Reply:
@howardjohn it makes sense, we can just remove that matching item from the webhook config.

Reply:
This is original issue to be addressed by adding `name` label - https://github.com/istio/istio/issues/10884 we need to ensure not break the original issue if just remove the `name` matchExpression from webhook config.

Reply:
@howardjohn & @morvencao I cannot find a good solution to handle the open issue in istio/istio. if we just remove `name` label matching from webhook config, it injects the sidecar into `istio-system` which leads sidecar pod cannot be started. any suggestions?

Reply:
The original PR istio/istio#10884 assumes that istio deployment is managed by tiller, and the CI test cases don't cover `sidecarInjectorWebhook.enableNamespacesByDefault=false` case, that's why we haven't found this issue.
IMO it's OK to merge this, wdyt? @howardjohn

Reply:
we can merge, but this script isn't the supported way to install istio for users, so this will mostly fix the tests not real users? As k believe users will either do helm template or kustomize. I think in kustomize we create the namespace and we set the label, so really only helm template is affected. The same problem exists on istio/istio as far as I know. I agree though, not sure we have a better solution as we can only match on labels.

Reply:
Yes. I found the same problem in istio/istio. cannot find a better solution so far.